pkg/password/manager.go

```diff
@@ -0,0 +1,33 @@
+package password
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+
+	"github.com/giantswarm/microerror"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type Manager interface {
+	GeneratePassword(length int) (string, error)
+	Hash(plaintext string) (string, error)
+}
+
+type SimpleManager struct {
+}
+
+func (m SimpleManager) GeneratePassword(length int) (string, error) {
+	bytes := make([]byte, length)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+func (m SimpleManager) Hash(plaintext string) (string, error) {
```

Comment:
We have an exported function `GeneratePassword` and `Hash`,
Can we add an exported function `GenerateHashedPassword` that calls them both?

Reply:
Probably yes, but i'd rather have separate functions (one function one purpose). @TheoBrigitte any thoughts?

Reply:
Also, we cannot because we need both the secret and the hash for different purposes (hash for basic auth, password for the agent to be able to access prometheus remote write endpoint)